huaweicloud/services/dataarts/resource_huaweicloud_dataarts_architecture_model.go

```diff
@@ -0,0 +1,356 @@
+				ForceNew: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"dw_type": {
+				Type:     schema.TypeString,
+				Default:  "UNSPECIFIED",
+				Optional: true,
+			},
+			"level": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"created_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"updated_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created_by": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"updated_by": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func buildCreateModelParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"name":        d.Get("name"),
+		"type":        d.Get("type"),
+		"description": utils.ValueIngoreEmpty(d.Get("description")),
+		"dw_type":     d.Get("dw_type"),
+		"level":       utils.ValueIngoreEmpty(d.Get("level")),
+		"is_physical": d.Get("physical"),
+	}
+	return bodyParams
+}
+
+func resourceArchitectureModelCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	createModelHttpUrl := "v2/{project_id}/design/workspaces"
+	createModelProduct := "dataarts"
+
+	modelClient, err := cfg.NewServiceClient(createModelProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DataArts Studio Client: %s", err)
+	}
+	createModelPath := modelClient.Endpoint + createModelHttpUrl
+	createModelPath = strings.ReplaceAll(createModelPath, "{project_id}", modelClient.ProjectID)
+	createModelOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders:      map[string]string{"workspace": d.Get("workspace_id").(string)},
+	}
+	createModelOpt.JSONBody = utils.RemoveNil(buildCreateModelParams(d))
+	createModelResp, err := modelClient.Request("POST", createModelPath, &createModelOpt)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	createModelBody, err := utils.FlattenResponse(createModelResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	model := utils.PathSearch("data.value", createModelBody, nil)
+	id, err := jmespath.Search("id", model)
+	if err != nil {
+		return diag.Errorf("error creating DataArts Studio architecture model: %s is not found in API response", "id")
+	}
+	d.SetId(id.(string))
+
+	return resourceArchitectureModelRead(ctx, d, meta)
+}
+
+func resourceArchitectureModelRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	readModelProduct := "dataarts"
+	modelClient, err := cfg.NewServiceClient(readModelProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DataArts Studio Client: %s", err)
+	}
+
+	if strings.Contains(d.Id(), "/") {
+		// if id contains name, it is for import
+		err := getModelByName(modelClient, d, region)
```

Comment:
`getModelByName` and `getModelById` should only return the model object and error.
then we  call check deleted and d.Set in the Read function